ReactNativeClient/lib/services/InteropService_Exporter_Md.js

```diff
@@ -31,36 +31,105 @@ class InteropService_Exporter_Md extends InteropService_Exporter_Base {
 		}
 	}
 
-	async replaceResourceIdsByRelativePaths_(item) {
-		const linkedResourceIds = await Note.linkedResourceIds(item.body);
-		const relativePath = rtrimSlashes(await this.makeDirPath_(item, '..'));
+	async findUniqueFilenameFromPathListAndFS(name, itemPaths) {
+		let counter = 1;
+
+		const relativeDir = dirname(name);
+		let nameNoExt = filename(name, true);
+		let extension = fileExtension(name);
+		if (extension) extension = `.${extension}`;
+		let nameToTry = nameNoExt + extension;
+
+		let filesInDir = await shim.fsDriver().readDir(`${this.destDir_}/${relativeDir}`);
+		filesInDir = filesInDir.map(fp => `${relativeDir}/${fp}`);
+		while (true) {
+			const exists = itemPaths.includes(nameToTry) || filesInDir.includes(nameToTry);
+			if (!exists) return nameToTry;
+			nameToTry = `${nameNoExt} (${counter})${extension}`;
+			counter++;
+			if (counter >= 1000) nameToTry = `${nameNoExt} (${new Date().getTime()})${extension}`;
+			if (counter >= 10000) throw new Error('Cannot find unique title');
+		}
+	}
```

Comment:
Is it not possible to use the existing logic with `friendlySafeFilename()` / `findUniqueFilename()` here? We could add a second "existingNames" parameter to `findUniqueFilename(name, existingNames = null)` and if presents it is used to check for existing filenames too. I'm not entirely sure it will replicate all the required logic so please confirm.

Reply:
Modified the existing `findUniqueFilename()` by adding a "reservedNames" parameter (I think it's slightly better name than "existingNames" because the files in the `reservedNames` are not yet created but are only reserved).